src/Sql/Sql/ManagedInstance/Cmdlet/SetAzureSqlManagedInstance.cs

```diff
@@ -197,12 +270,22 @@ protected override IEnumerable<AzureSqlManagedInstanceModel> ApplyUserInputToMod
             AzureSqlManagedInstanceModel existingInstance = ModelAdapter.GetManagedInstance(this.ResourceGroupName, this.Name);
             Management.Internal.Resources.Models.Sku Sku = new Management.Internal.Resources.Models.Sku();
 
-            if (Edition != null)
+            // Get current edition
+            string currentEdition = existingInstance.Sku.Tier;
+
+            // Get current hardware family
+            string currentComputeGeneration = existingInstance.Sku.Name.Contains(Constants.ComputeGenerationGen4) ? Constants.ComputeGenerationGen4 : Constants.ComputeGenerationGen5;
```

Comment:
This looks like it would cause problems when next compute gen is released. Please distinguish between Gen4, Gen5, and "unknown"

Reply:
I will use SqlSkuUtils.GetHardwareGenerationFromSkuName instead this if.